src/chapter02/exercises/exercise03.lean

```diff
@@ -7,13 +7,31 @@ def rational (x : ℝ) :=
 /-- The product of two rational numbers is always rational. -/
 lemma part_a : ∀ {a b : ℝ}, rational a → rational b → rational (a * b) :=
 begin
-  sorry
+  intros a b,
+  rintro ⟨x1,y1,h1⟩ ⟨x2,y2,h2⟩,
+  use [x1*x2, y1*y2],
+  simp,
+  ring_nf,
+  rw [mul_inv, h1, h2],
+  ring_nf,
 end
 
 /-- The product of two irrational numbers is not always irrational. -/
 lemma part_b : ¬ ∀ {a b : ℝ}, irrational a → irrational b → irrational (a * b) :=
 begin
-  sorry
+  simp, 
```

Comment:
This is a non-terminal `simp` which is bad; if people add other `simp` lemmas to mathlib then this proof might break. Why not use `push_neg` instead? That's what this tactic is for. Your proof might need some minor rearranging afterwards but nothing serious.